File: Day67/Part2/enemy_spawn.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int T;
    if (cin >> T) {
        vector<int> spawn(T);
        for (int i = 0; i < T; ++i) cin >> spawn[i];

        int max_spawn = -1;
        int skips = 0;
        bool skip_next = false;

        for (int i = 0; i < T; ++i) {
            if (skip_next) {
                skips++;
                skip_next = false;
                if (spawn[i] > max_spawn) {
                    max_spawn = spawn[i];
                }
                continue;
            }

            if (spawn[i] > max_spawn) {
                max_spawn = spawn[i];
                skip_next = true; 
            }
        }

        cout << skips << endl;
    }
    return 0;
}

File: Day67/Part2/enemy_spawn.py
def solve():
    try:
        line = input().split()
        if not line: return
        T = int(line[0])
        
        spawn = list(map(int, input().split()))
        
        max_spawn = -1
        skips = 0
        skip_next = False
        
        for i in range(T):
            if skip_next:
                skips += 1
                skip_next = False
                if spawn[i] > max_spawn:
                    max_spawn = spawn[i]
                continue
                
            if spawn[i] > max_spawn:
                max_spawn = spawn[i]
                skip_next = True
                
        print(skips)
    except EOFError:
        pass

if __name__ == "__main__":
    solve()
